### ISSUE TITLE
we need to handle non-flat contract directories

### ISSUE THREAD

**Issue by blinky3713** [f-o-a-m/purescript-web3-generator#35]
if you have the following project layout

```
contracts/
-- Foo.sol
-- SimpleStorage/
---- SimpleStorage.sol
```

then [chanterelle](https://github.com/f-o-a-m/chanterelle) or truffle generates the following

```
build/
-- contracts/
---- Foo.json
---- SimpleStorage/
------ SimpleStorage.json
```

If you assign `--abis` option to `build/contracts` then you will not get the SimpleStorage abi. We need to recursively build the modules and assign them the same relative hierarchy in the output dir.

**Comment by blinky3713** [comment]
fixed in #36 , the above example produces the following

```
-- src/
---- Contracts/
------ Foo.purs
------ SimpleStorage.purs
```

**Comment by safareli** [comment]
@blinky3713 if structure is:
```
contracts/
---- Foo.json
---- X/
------ Foo.json
```

shouldn't output be:
``` 
-- src/
---- Contracts/
------ Foo.purs
------ X/
-------- Foo.purs
```
?
